nexus/blueprint-execution/src/lib.rs

```diff
@@ -46,6 +49,31 @@ impl From<nexus_db_model::Sled> for Sled {
     }
 }
 
+// A vastly-restricted `Nexus` object that allows us access to some of Nexus
+// proper's capabilities.
+struct NexusContext<'a> {
+    opctx: OpContext,
+    datastore: &'a DataStore,
+}
+
+impl Base for NexusContext<'_> {
```

Comment:
> This really **isn't** the Nexus object, nor is it truly a subset, right? It's:
> 
> 1. Taking a chunk of code that used to be part of `struct Nexus`
> 2. Moving it out, and copying all fields from `struct Nexus` it used
> 3. ... But now it's a totally new object.

This is all correct. And in particular, in this case the logger we present is _different_ from the one stored in `Nexus`, which means any logs emitted by `plumb_service_firewall_rules` (or other `nexus-capabilities` methods it calls in turn) will use our RPW logger instead of the Nexus logger.
 
> I bring this up because in this use-case, you happen to use a single opctx + datastore, but there's a lot more stuff that **could** be used here by other capabilities, if we keep splitting this stuff up.
> 
> For example: `app/silo.rs` uses `self.external_resolver`, so presumably that field would need to be copied (or fully moved) into the `NexusContext` into a "capability to access silos".

All correct, yes.